src/batch.jl

```diff
@@ -0,0 +1,210 @@
+struct BatchIntegrand{F,Y,X}
+    # in-place function f!(y, x) that takes an array of x values and outputs an array of results in-place
+    f!::F
+    y::Vector{Y}
+    x::Vector{X}
+    max_batch::Int # maximum number of x to supply in parallel (defaults to typemax(Int))
+    function BatchIntegrand{F,Y,X}(f!::F, y::Vector{Y}, x::Vector{X}, max_batch::Int) where {F,Y,X}
+        max_batch > 0 || throw(ArgumentError("maximum batch size must be positive"))
```

Comment:
Shouldn't it just get the `max_batch` from `length(x)`, and require `length(x) == length(y)`?

Reply:
Though I guess this is okay since you call `resize!` below as needed.  And this way you can have `max_batch` be a large number without allocating lots of memory up-front.

In fact, it might be good to simply have a default `max_batch=typemax(Int)`.